lab03-jacob/lib/readfile.js

```diff
@@ -0,0 +1,56 @@
+var fs = require('fs');
+
+function readFiles(arg1, arg2, arg3, callback) {
```

Comment:
You're actually really close to a recursive implementation; given that you have all the duplicate code in each of these functions calling the next function, etc. If you want to give it a shot it would be an awesome exercise.